stores/multipart.go

```diff
@@ -37,7 +38,14 @@ type (
 		Size                uint64
 		DBMultipartUploadID uint      `gorm:"index;NOT NULL"`
 		Slabs               []dbSlice `gorm:"constraint:OnDelete:CASCADE"` // CASCADE to delete slices too
+	}
+
+	dbMultipartUploadUserMetadata struct {
```

Comment:
Is there a reason why you want to do 2 tables instead of 1 like we do for `dbSlab`?
Meaning that `dbUserMetadata` would have 2 columns where one is a foreign key on the object and the other one on the multipart upload and we swap them when we complete the multipart upload.

Saves us a table and makes completing an upload an "update" operation instead of an "insert+delete" which should be quite a bit faster.

Reply:
No reason, I didn't consider that option. I changed it to use the one table, and I definitely like that it gets rid of a whole table and of some duplication. The one thing I don't like is that we can't enforce the at least one of the foreign key fields is `NOT NULL` (XOR-like-constraint).

Reply:
Yeah we can't do that. The best we can do is have a unique index over those two columns. But that might not be worth it if we have proper testing in place.

Reply:
I have a unique index on those 3 fields (need to include the `key` too), I think that makes sense but I'm up for changing that. I'd rather add the constraint and remove it if necessary than the other way around.